flips/20220111-execution-effort.md

```diff
@@ -0,0 +1,555 @@
+# Variable Transaction Fees - Execution Effort I.
+
+| Status        | Proposed                                                                   |
+| :------------ | :------------------------------------------------------------------------- |
+| **FLIP #**    | [753](https://github.com/onflow/flow/pull/753) |
+| **Author(s)** | Janez Podhostnik (janez.podhostnik@dapperlabs.com)                      |
+| **Updated**   | 2022-01-13                                                                 |
+
+## Abstract
+
+This FLIP builds on the foundations of the [Variable Transaction fees FLIP](20211007-transaction-fees.md) and proposes a model for measuring the execution effort of transactions by choosing certain functions/operations, that are called during the execution of a transaction, to have a related execution effort cost. This FLIP explores a choice of functions/operations and uses data collected from sample transactions and a linear model fit to determine the cost of each chosen functions/operations, so that on average the execution effort of a transaction is proportional to the execution time of the transaction. This FLIP also explores the FLOW cost of a unit of execution effort.
+
+## Objective
+
+In the [Variable Transaction fees FLIP](20211007-transaction-fees.md) the transaction execution fees are defined as the part of the transaction fees that account for the resources (bandwidth, computing power) needed to execute the transactions' script, to verify the transaction execution and to handle the propagation of transaction execution results. The execution fees (<img style="transform: translateY(0.1em); background: white;" src="./20220111-execution-effort/eq/A0hSAHHSW2.svg">) are defined as a execution effort cost function (<img style="transform: translateY(0.1em); background: white;" src="./20220111-execution-effort/eq/sCA7cYTn4j.svg">) of the execution effort (<img style="transform: translateY(0.1em); background: white;" src="./20220111-execution-effort/eq/WlYFrmB6Y8.svg">) of the transaction <img style="transform: translateY(0.1em); background: white;" src="./20220111-execution-effort/eq/Uzsd7p4YBJ.svg">.
+
+The aim of this FLIP is to create a model for measuring the execution effort of transactions, that satisfies the following criteria:
+
+- The model must be better then the current model of measuring the execution effort.
+- The model must be straightforward to understand and implement.
+- The implementation of the model must not significantly degrade the execution time of transactions.
+- There must be a clear improvement path for the model.
```

Comment:
I think there is one very important requirement missing:
> * The model must be byzantine compatible, i.e. hard to game / circumvent. 

This is probably a criterium that is hard to completely enforce, but I think we need a reasonable level of confidence here.